jubatus/core/bandit/epsilon_decreasing.hpp

```diff
@@ -0,0 +1,86 @@
+// Jubatus: Online machine learning framework for distributed environment
+// Copyright (C) 2015 Preferred Networks and Nippon Telegraph and Telephone Corporation.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License version 2.1 as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+#ifndef JUBATUS_CORE_BANDIT_EPSILON_DECREASING_HPP_
+#define JUBATUS_CORE_BANDIT_EPSILON_DECREASING_HPP_
+
+#include <string>
+
+#include "jubatus/util/data/serialization.h"
+#include "jubatus/util/data/optional.h"
+#include "bandit_base.hpp"
+#include "summation_storage.hpp"
+#include "jubatus/util/math/random.h"
+
+namespace jubatus {
+namespace core {
+namespace bandit {
+
+class epsilon_decreasing : public bandit_base {
+ public:
+  struct config {
+    bool assume_unrewarded;
+    double d;
```

Comment:
I don't think `d` is good for the parameter name because users may not imagine the meaning of the parameter.
How about `decreasing_rate` or something like that?